### ISSUE TITLE
ERROR OF SCTransform: Error in h(simpleError(msg, call)) :   error in evaluating the argument 'x' in selecting a method for function 'as.matrix': invalid character indexing

### ISSUE THREAD

**Issue by jiangsiao** [satijalab/seurat#6810]
Error in h(simpleError(msg, call)) :
  error in evaluating the argument 'x' in selecting a method for function 'as.matrix': invalid character indexing
###########################

**> HCC04T.SCT <- SCTransform(HCC04T, vars.to.regress = c("percent.mito","nFeature_RNA","nCount_RNA"), method = "glmGamPoi")**

_Calculating cell attributes from input UMI matrix: log_umi
Variance stabilizing transformation of count matrix of size 0 by 4
Model formula is y ~ log_umi
Get Negative Binomial regression parameters per gene
Using 0 genes, 4 cells
  |                                                                      |   0%
Error in h(simpleError(msg, call)) :
  error in evaluating the argument 'x' in selecting a method for function 'as.matrix': invalid character indexing_

**> HCC04T**
_An object of class Seurat
28450 features across 4 samples within 2 assays
Active assay: integrated (2000 features, 2000 variable features)
 1 other assay present: RNA
 3 dimensional reductions calculated: pca, umap, tsne_

Maybe the false is in count matrix, because the execution prompt says there are no genes in this matrix. How can I solve it?

**Comment by saketkc** [comment]
It seems your `HCC04T's` RNA assay has no UMIs`. What is the output of 

```r
x <- GetAssayData(HCC04T, assay="RNA", slot="counts")` 
x[1:5, 1:5]
```

**Comment by SSSJe** [comment]
Yes. I found out there was something wrong with the data later.  Thx!
The links of example is below:
https://github.com/satijalab/seurat/issues/4877

**Comment by kollo97** [comment]
For anyone stumbling over this issue, the sctransform funtion from vst has a parameter 'min_cells' set to 5 by default [https://rdrr.io/cran/sctransform/man/vst.html](https://rdrr.io/cran/sctransform/man/vst.html).
I was working with meta cells and thus had a number of cells lower than 5. 
Setting Seurat::SCTransform(se, min_cells = 1) solved the error for me.
